src/test/java/com/fryrank/dal/ReviewDALTests.java

```diff
@@ -0,0 +1,51 @@
+package com.fryrank.dal;
+
+import com.fryrank.model.Review;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReviewDALTests {
+    @Mock
+    MongoTemplate mongoTemplate;
+
+    @InjectMocks
+    ReviewDALImpl reviewDAL;
+
+    @Test
+    public void testGetAllReviewsByRestaurantId() throws Exception {
+        Query query = new Query();
+        query.addCriteria(Criteria.where("restaurantId").is("1"));
+        List<Review> expectedReviews = new ArrayList<>() {
+            {
+                add(new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1"));
+                add(new Review("review_id_2", "restaurant_id_2", "author_id_2", 7.0 , "title_2", "body_2"));
+            }
+        };
+        Mockito.when(mongoTemplate.find(query, Review.class)).thenReturn(expectedReviews);
```

Comment:
Recommend importing when method specifically instead of entire Mockito

Reply:
Done in b1cd8234a3db8d047ba0a0127860fc45e407be8a